crates/tools/fabric-gen/src/gen.rs

```diff
@@ -62,16 +62,42 @@ impl TypeEntry {
                 let te = TypeEntry::from_type_reader(t, r, attr);
                 (te.name, te.ns, format!("{}{}", q, te.prefix)) // te prefix is empty always
             }
-            Type::HRESULT => (
-                "HRESULT".to_string(),
-                "::windows_core".to_string(),
-                empty_str,
-            ),
-            Type::GUID => ("GUID".to_string(), "::windows_core".to_string(), empty_str),
-            Type::TypeRef(x) => {
-                let te = TypeEntry::from_typeref(x, r);
-                (te.name, te.ns, te.prefix)
+            Type::Name(n) => {
+                match n {
+                    &TypeName::HRESULT | &TypeName::HResult => (
+                        "HRESULT".to_string(),
+                        "::windows_core".to_string(),
+                        empty_str,
+                    ),
+                    &TypeName::GUID => {
+                        ("GUID".to_string(), "::windows_core".to_string(), empty_str)
+                    }
+                    &TypeName::PWSTR => {
+                        let t = match attr {
+                            ParamAttributes::In => "PCWSTR",
+                            _ => "PWSTR",
+                        };
+                        (t.to_string(), "::windows_core".to_string(), empty_str)
+                    }
+                    // TODO: this might not be right?
+                    &TypeName("Windows.Win32.Foundation", "BOOLEAN") => (
+                        "BOOLEAN".to_string(),
+                        "::windows_core".to_string(),
+                        empty_str,
+                    ),
+                    _ => panic!("typename not supported: {} {}", n.namespace(), n.name()),
+                }
             }
+            // Type::HRESULT => (
```

Comment:
Remove commented code?

Reply:
file removed from the generator removal PR.